pkg/controller/migration/convert/network.go

```diff
@@ -0,0 +1,165 @@
+package convert
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	operatorv1 "github.com/tigera/operator/pkg/apis/operator/v1"
+	v1 "github.com/tigera/operator/pkg/apis/operator/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	containerCalicoNode = "calico-node"
+	containerInstallCNI = "install-cni"
+)
+
+func handleNetwork(c *components, install *Installation) error {
+	// TODO: be more elegant about creation of this spec field if it already exists
+	if install.Spec.CalicoNetwork == nil {
+		install.Spec.CalicoNetwork = &operatorv1.CalicoNetworkSpec{}
+	}
+
+	// CALICO_NETWORKING_BACKEND
+	netBackend, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "CALICO_NETWORKING_BACKEND")
+	if err != nil {
+		return err
+	}
+	if netBackend != nil && *netBackend != "" && *netBackend != "bird" {
+		return ErrIncompatibleCluster{"only CALICO_NETWORKING_BACKEND=bird is supported at this time"}
+	}
+
+	// Calico CNI
+	if c.calicoCNIConfig == nil {
+		return fmt.Errorf("no 'calico' cni conf found in CNI_NETWORK_CONFIG on install-cni")
+	}
+
+	// FELIX_DEFAULTENDPOINTTOHOSTACTION
+	defaultWepAction, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "FELIX_DEFAULTENDPOINTTOHOSTACTION")
+	if err != nil {
+		return err
+	}
+	if defaultWepAction != nil && strings.ToLower(*defaultWepAction) != "accept" {
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("unexpected FELIX_DEFAULTENDPOINTTOHOSTACTION: '%s'. Only 'accept' is supported.", *defaultWepAction),
+		}
+	}
+
+	// IP
+	ipMethod, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "IP")
+	if err != nil {
+		return err
+	}
+	if ipMethod != nil && strings.ToLower(*ipMethod) != "autodetect" {
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("unexpected IP value: '%s'. Only 'autodetect' is supported.", *ipMethod),
+		}
+	}
+
+	// IP_AUTODETECTION_METHOD
+	am, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "IP_AUTODETECTION_METHOD")
+	if err != nil {
+		return err
+	}
+	if am != nil {
+		tam, err := getAutoDetection(*am)
+		if err != nil {
+			return err
+		}
+		install.Spec.CalicoNetwork.NodeAddressAutodetectionV4 = &tam
+	}
+
+	// CNI portmap plugin
+	if _, ok := c.pluginCNIConfig["portmap"]; ok {
+		hp := v1.HostPortsEnabled
+		install.Spec.CalicoNetwork.HostPorts = &hp
+	} else {
+		hp := v1.HostPortsDisabled
+		install.Spec.CalicoNetwork.HostPorts = &hp
+	}
+
+	// CNI bandwidth plugin
+	if _, ok := c.pluginCNIConfig["bandwidth"]; ok {
```

Comment:
I think this will block upgrades for now. I think we should just be ignoring this for now.